lib/dao.js

```diff
@@ -1562,7 +1565,23 @@ DataAccessObject._coerce = function(where) {
 
       continue;
     }
+
     var DataType = props[p] && props[p].type;
+    if (p.match(/\./)) {
+      var model = p.split('.')[0];
+      var prop = p.split('.').slice(1);
+
+      if (props[model]) {
+        var clause = {};
+        clause[prop] = where[p];
+        where[p] = Array.isArray(props[model].type) ?
+          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
```

Comment:
What if `props[model]` is an array of scalar values, e.g. `{ type: ['string'] }`? Please add a unit test for this case, e.g.

```js
User.find({where: {'tags.prop': true}},...)
```